### PR TITLE
Update for Seldon 1.0.1 for Kubeflow 1.0 Release and add CertManager to kfdefs

### PR DESCRIPTION
**Pull request by cliveseldon**
Update for Seldon 1.0.1

* Add cert-manager to all KFDefs ; cert-manager will be used to provision self signed certificates for various components

Related to #737 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/717)

### CODE REVIEW

**Comment by cliveseldon** [comment]
@kkasravi @yanniszark How does one test this PR update with a kubeflow install for the 1.0 kubeflow release?

**Comment by cliveseldon** [comment]
Ok I have tested with latest kfctl and these manifests in the PR @kkasravi @yanniszark

File: seldon/seldon-core-operator/base/resources.yaml [hunk 1]
```diff
@@ -0,0 +1,3656 @@
...
...WxqY3kxallUQ0NBU0l3CkRRWUpLb1pJaHZjTkFRRUJCUUFEZ2dFUEFEQ0NBUW9DZ2dFQkFKazQ1eW9DRXRMLzZZVWM1bVovZ0o2WWUvVUUKMDZWT1U5Rk5iWk11eDM3VTNlUzdDR3N2aExCSVU1eFduSElMR1VYUTNyTk9kQTZHYXRqamsydkowTTU4TFh2KwpKZzF1MTYzSjhCSi9hSjUxRmhPTUVmSFBpWDZvSTlYUTJEWlpZUUIvV3RpTHJRazlXS2Zld1lzREIvNzBrUnNVCk0yMm1lMkJSR0cvb2gvcnY0M04wYmtodHA5QUVQd2NkWFZOcmJiWHVuRytpZEJZU0FrOUVPTXFwZXVRRDJEV1cKc3g1TmJXakNHY2YzZ1pvNjEyZCtpUysyQVNYMk1TM0pQOHppS1N1VjVQMVhTRk5qUS9peHh3S0dkVjdzWkxUMAoreFdDSXNZY0cxSXc5SXU4MDRXSVkrdWpTM20rYXliYS9ESmx0eUJ5STlSclErME8zOFQ0M2h4bGgzc0NBd0VBCkFhTkNNRUF3RGdZRFZSMFBBUUgvQkFRREFnS2tNQjBHQTFVZEpRUVdNQlFHQ0NzR0FRVUZCd01CQmdnckJnRUYKQlFjREFqQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01BMEdDU3FHU0liM0RRRUJDd1VBQTRJQkFRQ1FsRXZ1RDVSSwozMENCR1FBenk3NGtacmtDZEdRZFBGOGRBalN4VVVJem1EdFBJWVZHTFF0RFpPTlphcmJJZmVlUVRUeStIQXJGCkVlb1g0ajVGNVlNaURUQ2FGOWgvME5EbGVmSDFmWGNsOHpvaytCQkpIbVlSMklhNVRDVk5iRjcrYXl0TkowclIKLzV3QlViYlkzR1ZoZWp4cHVOSzNXaFZEMjVZdWR0TTVPZUdZcXJjcjJSbHd5REYrd0pmWXJteFBtd1BneTBJNwpGTW02ZXhGSGFTUDYydlNUOFI5M3FQSzJiTllod0lyclcvLzN4OWRSdnF6dmFoUUxtenlvUDBBL2JBK3ZZNGhHCkZJbmhmOGtFdW5iZTh0SXk5bmVDejlhYklZUVNwQm16UFpVZG5rS2xiRlVGTVFpbkVobU1JdzY0Y0RvWXBPUW4KL2NVbXhDQlB5KzZrCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K'
```
Comment (jlewi) [inline seldon/seldon-core-operator/base/resources.yaml]: Do you really want to check the certificate into source control? Is it possible to use certmanager to generate it?

**Comment by jlewi** [comment]
See #711 - If seldon infra (e.g. the operator, webhook etc...) isn't compatible with the ISTIO side car then you may need to add the label to pods to prevent istio side car injection.

You can check with @richardsliu to see if anything needs to be added to the Application CR to work with upgrades.

**Comment by jlewi** [comment]
The v0-1-branch has also been cut so after this is merged you will want to cherry pick it.

File: seldon/seldon-core-operator/base/resources.yaml (hunk 1 above)
Comment (cliveseldon) [reply seldon/seldon-core-operator/base/resources.yaml]: Yes. This is just a dev certificate. I can change so we assume certmanager is available.

**Comment by cliveseldon** [comment]
> See #711 - If seldon infra (e.g. the operator, webhook etc...) isn't compatible with the ISTIO side car then you may need to add the label to pods to prevent istio side car injection.

I don't think this should cause an issue as we don't use stateful sets for the operator but can check.

> 
> You can check with @richardsliu to see if anything needs to be added to the Application CR to work with upgrades.

@richardsliu If there are particular links needed for upgrades can you refer them to me

File: seldon/seldon-core-operator/base/resources.yaml (hunk 1 above)
Comment (cliveseldon) [reply seldon/seldon-core-operator/base/resources.yaml]: Actually I see the cert-manager is only in kfctl_istio_dex.yaml - so what would be the preferred way of provding certs for the other options? Should cert-manager not be in all? @jlewi @richardsliu

**Comment by jlewi** [comment]
> I don't think this should cause an issue as we don't use stateful sets for the operator but can check.

@cliveseldon Its not just stateful sets. Its if you have any pods that won't work with ISTIO side car injection.

It looks like #711 added the label to a bunch of Seldon resources. This PR is overwriting that change.  I only wanted to call your attention to that so you could determine whether the label to disable side car injection needs to be added to any Seldon resources.

File: seldon/seldon-core-operator/base/resources.yaml (hunk 1 above)
Comment (cliveseldon) [reply seldon/seldon-core-operator/base/resources.yaml]: I have added cert-manager to all kfdefs as part of this PR

**Comment by cliveseldon** [comment]
@krishnadurai Can you have a look at this

**Comment by yuzisun** [comment]
+1 to make cert manager as part of standard kubeflow installation, KFServing 0.3 will assume cert managed is installed.

**Comment by yanniszark** [comment]
Regarding cert-manager, I believe that we will soon (if not now) need to provision many certificates for admission controllers in Kubeflow.
The best way I know to solve this problem is cert-manager.
In addition, kubebuilder relies on cert-manager for that problem.
In the v0.7 existing_arrikto config, we use cert-manager for the PodDefault admission webhook and thanks to that, we got rid of the admission-bootstrap job that was generating the cert.
@jlewi what do you think? Should cert-manager be something Kubeflow relies on?

**Comment by jlewi** [comment]
/lgtm
/approve

@cliveseldon note this change will need to be cherry-picked on to the 1.0-branch. So I updated the PR description to avoid closing #737

**Comment by cliveseldon** [comment]
Same failures as happening for for #651 
Could this be a test flake or any ideas @jlewi

**Comment by jlewi** [comment]
@cliveseldon see comment
https://github.com/kubeflow/manifests/pull/651#issuecomment-579218860

Since this PR is modifying KFDef it probably shouldn't be opting out of testing.

**Comment by cliveseldon** [comment]
> @cliveseldon see comment
> [#651 (comment)](https://github.com/kubeflow/manifests/pull/651#issuecomment-579218860)
> 
> Since this PR is modifying KFDef it probably shouldn't be opting out of testing.

I didn't realise I changed anything to opt out of tests. The issue is a failing test forkatib and endpoint-controller as a symptom 

```
E     kubeflow.testing.util.TimeoutError: Timeout waiting for deployment cloud-endpoints-controller in namespace kubeflow
../../../../../../testing/py/kubeflow/testing/util.py:453: TimeoutError
```

see [here](http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-manifests-presubmit-kfctl-go-iap-717-3a843dc-5552-7697?tab=workflow&nodeId=kubeflow-manifests-presubmit-kfctl-go-iap-717-3a843dc-5552-7697-81458991&sidePanel=logs%3Akubeflow-manifests-presubmit-kfctl-go-iap-717-3a843dc-5552-7697-81458991%3Amain)

Can anyone decipher this? How do I access the stackdriver logs?

**Comment by jlewi** [comment]
@cliveseldon apologies in my comment on the KFSeving PR one of the things I proposed was opting out of tests. What I meant is that that wouldn't be a suitable option on this PR.

The timeout issue looks like kubeflow/kubeflow#4693 this is fixed on master. So you should rebase off the latest master and update the branch.

This should pull in the fix.

If the katib test is still flaky then we can disable it but I believe that issue might also be affecting katib.

**Comment by jlewi** [comment]
/lgtm
/approve

@cliveseldon Woo Hoo! 

Don't forget to submit a cherrypick 
https://github.com/kubeflow/kubeflow/blob/master/hack/cherry-picks.md

**Comment by jlewi** [comment]
@cliveseldon actually hold off cherry-picking this on the v1.0-branch. I'm working on some changes that might make that simpler.

**Comment by cliveseldon** [comment]
@jlewi ok sure

**Pull request merged**